### DISCUSSION TITLE
Multi-graph in One Sample.

### DISCUSSION THREAD

**Discussion by matcyr** [pyg-team/pytorch_geometric#6569] [General]
Hi, I am recently trying to implement a regression model based on multi-graph data interaction.
I would like to have multi-graphs saved in one sample and applied mini-batch. I found the example of having a pair of graphs here: https://pytorch-geometric.readthedocs.io/en/latest/notes/batching.html. 
However, I am having 7 graphs together for one sample. Do I have a better way rather than using the 'follow_batch'? 

Thanks,
Yurui

**Comment by wsad1** [comment]
Ah, i guess the problem you want to avoid is is you'll have to specify seven variabes for `x` and `edge_index` in your `Data` object and then pass 7 `x`s to `follow_batch`. Unfortunately at the moment `follow_batch` is the only way to do this.
You could use `for` loops and `setattr` in `__init__` to make your code shorter
```
SevenGraphs(Data):
    def __init__(list_x, list_edge_index) # list of 7 node features and edge indicies
        for i in range(7):
            setattr(self, f'x_{i}', list_x[i]) 
``
